parser/prism/Translator.cc

```diff
@@ -682,19 +682,22 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             // These 2 need to be called on a new Translator with isInMethod set to true
             Translator childContext = enterMethodDef();
 
-            unique_ptr<parser::Node> params;
+            auto receiver = translate(defNode->receiver);
+            auto body = childContext.translate(defNode->body);
 
-            // The definition has no parameters but still has parentheses, e.g. `def foo(); end`
-            // In this case, Sorbet's legacy parser will still hold an empty Args node, so we need to
-            // create one here
-            if (defNode->parameters == nullptr && rparenLoc.start != nullptr) {
-                params = make_unique<parser::Args>(location, NodeVec{});
+            unique_ptr<parser::Args> params;
+            if (defNode->parameters != nullptr) {
+                params = childContext.translateParametersNode(defNode->parameters);
             } else {
-                params = childContext.translate(up_cast(defNode->parameters));
```

Comment:
This `translate` call would always hit the `PM_PARAMETERS_CASE`, but would return a general `unique_ptr<parser::Node>` result. Rather than casting, it was nicer to just extract a function that takes exactly a `pm_parameters_node *` and returns a `unique_ptr<parser::Args>`.